### PR TITLE
fix: make a valid highlight link required to post a highlight

### LINKED ISSUE #2560: Bug: No clear error message/state shown when posting a highlight that is too long
### Describe the bug

When posting a highlight that exceeds 500 characters, the input does not show a clear error message and the border does not show an error state. The only feedback given is the count in the corner turns red. This is not that visible and not obvious. 

Screenshot showing this: 
Screenshot 2024-01-31 at 00 39 22

### Steps to reproduce

1. Click on Highlights tab
2. Click on the post a highlight input
3. Add text that exceeds the 500 limit
4. Notice the input shows no clear error except the count turns red

### Browsers

_No response_

### Additional context (Is this in dev or production?)

localhost

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by FatumaA**
## Description
This PR adds a clear error border and message when the highlight is too long. 

Notes: 
I went with adding an extra error span to the existing one so that the too-long error does not block the other errors like the "wrong type of highlight" error when linking an issue.
This is shown in the screenshots below.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
Fixes #2560 

## Mobile & Desktop Screenshots/Recordings

Screenshot 2024-02-05 at 14 34 36

Screenshot 2024-02-05 at 14 34 47

## Steps to QA
1. Click on the "add a highlight to show your work!" input
2. Paste a description that is longer than 500 characters
3. Notice the only feedback is the char count at the bottom right corner of the text box
4. With this PR, notice the border of the text box turns red and there is a msg displayed at the top of the textbox clearly explaining what the error is

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by jpmcb — APPROVED** [review]
Approved the netlify build. Looks solid 👍🏼

File: components/molecules/HighlightInput/highlight-input-form.tsx [hunk 1]
```diff
@@ -541,12 +541,21 @@ const HighlightInputForm = ({ refreshCallback }: HighlightInputFormProps): JSX.E
           <form onSubmit={handlePostHighlight} className="flex flex-col gap-4 font-normal">
             <p role="alert">
               {errorMsg && (
-                <span className="inline-flex items-center gap-2 px-2 py-1 text-red-500 bg-red-100 border border-red-500 rounded-md w-full text-sm">
+                <span className="inline-flex items-center gap-2 px-2 py-1 mt-2 text-red-500 bg-red-100 border border-red-500 rounded-md w-full text-sm">
                   <MdError size={20} /> {errorMsg}
                 </span>
               )}
+              {!hasValidCharCount && (
+                <span className="inline-flex items-center gap-2 px-2 py-1 mt-2 text-red-500 bg-red-100 border border-red-500 rounded-md w-full text-sm">
```
Comment (nickytonline) [inline components/molecules/HighlightInput/highlight-input-form.tsx:549]: @FatumaA, the colors being used cause colour contrast issues.

CleanShot 2024-02-05 at 07 57 14

CleanShot 2024-02-05 at 07 58 42

`bg-red-100` should be white instead.

```suggestion
                <span className="inline-flex items-center gap-2 px-2 py-1 mt-2 text-red-500 bg-white border border-red-500 rounded-md w-full text-sm">
```

**Review by takanome-dev — APPROVED** [review]
Looks great 👍 🍕

**Review by NsdHSO — APPROVED** [review]

**Comment by FatumaA** [comment]
> There was another issue that fixed the negative count like you have in your screenshot.
> 
> CleanShot 2024-02-08 at 08 20 08@2x
> 
> Because of that fix, we no longer need to provide a message that your highlight needs to be 500 characters or less because you won't be able to type or past more than 500 characters now.
> 
> See #2613

Ok, so I made the following changes: 
- removed the hasValidCharCount from that file - it will no longer be needed
- removed the error message and error border - they won't be needed

I'm now working around: 
- getting the error messages from the auto-summarize button to clear when the incorrect highlight is replaced with a correct highlight so that the Post button 

I will mark this as draft till I get that behaviour consistent

**Comment by FatumaA** [comment]
Hi,
This is ready for review. 
I made this changes:
- removed the hasValidCharCount from that file - it will no longer be needed
- removed the error message and error border - they won't be needed
- got the error messages from the auto-summarize button to clear when the incorrect highlight is replaced with a correct highlight 
- got the add highlight link input to check if link is valid and set error if not
- Post button is disabled when the body text is empty and when the highlight link URL is invalid

Please take a look. 

Thanks!

Video to illustrate changes: 


https://github.com/open-sauced/app/assets/67555014/06550550-3627-4bcb-a072-3a38ce0d6157

**Comment by brandonroberts** [comment]
Thanks @FatumaA, one minor change is to clear the error if the field is empty. Everything else looks good

**Comment by FatumaA** [comment]
> Thanks @FatumaA, one minor change is to clear the error if the field is empty. Everything else looks good

Hi, 
Done 😌


https://github.com/open-sauced/app/assets/67555014/a08533cf-5c14-4539-ab36-d316d0cdb009

**Review by nickytonline — APPROVED** [review]
:shipit:

**Pull request merged**